src/netaddress.cpp

```diff
@@ -778,31 +780,34 @@ std::vector<unsigned char> CNetAddr::GetGroup(const std::vector<bool> &asmap) co
         return vchRet;
     }
 
-    vchRet.push_back(net_class);
+    vchRet.push_back(address.GetNetClass());
+    int nStartByte{0};
     int nBits{0};
 
-    if (IsLocal()) {
+    if (address.IsLocal()) {
         // all local addresses belong to the same group
-    } else if (IsInternal()) {
-        // all internal-usage addresses get their own group
+    } else if (address.IsInternal()) {
+        // All internal-usage addresses get their own group.
+        // Skip over the INTERNAL_IN_IPV6_PREFIX returned by CAddress::GetAddrBytes().
+        nStartByte = INTERNAL_IN_IPV6_PREFIX.size();
```

Comment:
nit: i think it would also be possible to keep the prefix and set `nBits = (INTERNAL_IN_IPV6_PREFIX.size() + ADDR_INTERNAL_SIZE) * 8;`. Then we wouldn't need to reintroduce `nStartByte`, which was previously also used for other address types.